src/borg/cache.py

```diff
@@ -70,6 +70,10 @@ def __init__(self, repository, key, manifest, path=None, sync=True, do_files=Fal
         self.key = key
         self.manifest = manifest
         self.path = path or os.path.join(get_cache_dir(), repository.id_str)
+        if os.environ.get('BORG_UNIQ_HOSTNAME'):
```

Comment:
the hostname is also the same after cloning, right?
but maybe it is remembered more easily to be changed than some borg uuid. hmm...